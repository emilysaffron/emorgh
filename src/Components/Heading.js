import Grid from "@bbc/psammead-grid";
import { Headline } from "@bbc/psammead-headings";
import getGridGroups from "../helpers/GetGridGroups";

const Heading = ({ text }) => {
  return (
    <Grid
      item
      startOffset={getGridGroups([2, 2, 2, 2, 2, 2])}
      columns={getGridGroups([8, 8, 8, 8, 8, 8])}
    >
      <Headline>{text}</Headline>
    </Grid>
  );
};

export default Heading;
