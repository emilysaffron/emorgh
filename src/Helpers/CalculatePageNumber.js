const CalculatePageNumber = (currentPageNumber, direction) => {
  let pageNumber = currentPageNumber;
  if (direction === "Next" && currentPageNumber < 5) {
    pageNumber += 1;
  }

  if (direction === "Previous" && currentPageNumber > 1) {
    pageNumber -= 1;
  }

  return pageNumber;
};

export default CalculatePageNumber;
