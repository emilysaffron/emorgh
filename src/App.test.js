import add from "./Function";
import CalculatePageNumber from "./Helpers/CalculatePageNumber";
import renderer from "react-test-renderer";
import Heading from "./Components/Heading";
test("faketest", () => {
  expect(true).toBeTruthy();
});

test("add", () => {
  expect(add(1, 2)).toBe(3);
});

//Unit Test

test("CalculatePageNumber", () => {
  expect(CalculatePageNumber(3, "Next")).toBe(4);
  expect(CalculatePageNumber(3, "Previous")).toBe(2);
  expect(CalculatePageNumber(5, "Next")).toBe(5);
  expect(CalculatePageNumber(1, "Previous")).toBe(1);
});

//Snapshot Test
it("Heading Snapshot", () => {
  const tree = renderer.create(<Heading text="text" />).toJSON();
  expect(tree).toMatchSnapshot();
});
