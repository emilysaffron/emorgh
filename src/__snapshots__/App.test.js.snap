// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Snapshot 1`] = `
<div
  className="css-vxrtof-GridComponent e57qer20"
  dir="ltr"
>
  <h1
    className="css-1omfudl-Headline e14hemmw0"
  >
    text
  </h1>
</div>
`;
